### DISCUSSION TITLE
simulation error when simulate data after training SINDy

### DISCUSSION THREAD

**Discussion by BK201-kkk** [dynamicslab/pysindy#504] [Q&A]
here is my code:
optimizer = ps.STLSQ(threshold=0.00001, alpha=0, max_iter=1000)
library = ps.PolynomialLibrary(**degree=3**, include_bias=False)
model = ps.SINDy(optimizer=optimizer, feature_library=library)
model.fit(normalized_coefs_i.T, t=dt2)
**tsim =np.linspace(0,1000,15000)**
x_sim2 = model.simulate(normalized_coefs_i.T[200, :], tsim)

I have a set of 8-dimensional data with 4000 time steps. I used interpolation method to make it into 8-dimensional data with 12000 time points (the time range is still 0-4000). A third-order SINDy model was trained. However, in the simulate step, it is difficult to train data with a time range exceeding 650. If you increase the simulate time, overflow will occur(this problem does not occur in the second-order model). I suspect that my wrong choice of step size is causing the numerical problem. Do you know the integration method used by psindy so that I can calculate the appropriate step size based on the relevant formula? Or do you have another opinion on this issue? Thank you so much!
